src/components/RegistrationForm.js

```diff
@@ -0,0 +1,105 @@
+import React, { PropTypes } from 'react';
+import ReactDOM from 'react-dom';
+import styled from 'styled-components';
+import SocialRegister from './SocialRegister';
+import '../App.css';
+
+export default class RegistrationForm extends React.Component {
+    state = {
+        agreeNotChecked: true,
+        emailIsEmpty: true,
+        nickIsEmpty: true,
+        passwordIsEmpty: true
+    };
+    static propTypes = {
+        email: PropTypes.string.isRequired,
+        nick: PropTypes.string.isRequired,
+        password: PropTypes.string.isRequired
+    };
+
+    onBtnClickHandler = (e) => {
+        e.preventDefault();
+        let email = ReactDOM.findDOMNode(this.refs.email).value;
+        let nick = ReactDOM.findDOMNode(this.refs.nick).value;
+        alert(email + '\n' + nick);
+    }
+
+    onCheckAgreeClick = (e) => {
+        e.preventDefault();
+        this.setState(prevState => ({agreeNotChecked: !prevState.agreeNotChecked}));
+    }
+
+    onFieldChange = (fieldName, e) => {
+        if  (e.target.value.trim().length > 0)  {
+            this.setState({['' + fieldName]: false});
```

Comment:
looked in the example, I didn't quite understand why this was implemented, but it worked and I left it